projects/igniteui-angular/src/lib/grids/grid-base.directive.ts

```diff
@@ -7142,7 +7159,7 @@ export abstract class IgxGridBaseDirective extends DisplayDensityBase implements
         const fields = this.generateDataFields(data);
         const columns = [];
 
-        fields.forEach((field) => {
+        fields.filter((field) => !this.autoGenerateExclude.includes(field)).forEach((field) => {
```

Comment:
@ChronosSF Any reason this is not baked into `generateDataFields()` called above? Since that's where all the field filtering happens in overrides for Tree/HGrid too, sort of feels like that's where it belongs

Reply:
Because I am blind, though it would work the same either way.